Mathlib/SetTheory/Cardinal/Finite.lean

```diff
@@ -177,13 +177,13 @@ theorem card_of_subsingleton (a : α) [Subsingleton α] : Nat.card α = 1 := by
   letI := Fintype.ofSubsingleton a
   rw [card_eq_fintype_card, Fintype.card_ofSubsingleton a]
 
--- @[simp] -- Porting note (#10618): simp can prove this
-theorem card_unique [Unique α] : Nat.card α = 1 :=
-  card_of_subsingleton default
-
 theorem card_eq_one_iff_unique : Nat.card α = 1 ↔ Subsingleton α ∧ Nonempty α :=
   Cardinal.toNat_eq_one_iff_unique
 
+@[simp]
+theorem card_unique [Nonempty α] [Subsingleton α] : Nat.card α = 1 := by
```

Comment:
How come `simp` can't prove this anymore?

Reply:
The old proof goes via `Fintype.card` and [Unique.fintype](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Data/Fintype/Basic.html#Unique.fintype) (which shouldn't be an instance, because it's not clear whether it creates a diamond with instances for `Prod` and pi types).